include/openenclave/advanced/allocator.h

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+/**
+ * @file allocator.h
+ *
+ * This file defines the interface that pluggable allocators must implement.
+ * See
+ * https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/PluggableAllocators.md
+ *
+ */
+
+#ifndef OE_ADVANCED_ALLOCATOR_H
+#define OE_ADVANCED_ALLOCATOR_H
+
+#include <openenclave/bits/types.h>
+
+/**
+ * @cond IGNORE
+ */
+OE_EXTERNC_BEGIN
+
+/**
+ * @endcond
+ */
+
+/**
+ * Callback for initializing the allocator.
+ *
+ * This function will be called by oecore at an appropriate time during enclave
+ * initialization. The allocator is expected to perform any global
+ * initialization in this function. This function is called prior to any global
+ * initialization.
+ *
+ * @param[in] heap_start_address The starting address of the enclave heap to be
+ * managed by the allocator.
+ * @param[in] heap_end_address The ending address of the enclave heap.
+ */
+void oe_allocator_init(void* heap_start_address, void* heap_end_address);
+
+/**
+ * Callback for cleaning up the allocator.
+ *
+ * This function will be called by oecore at an appropriate time during enclave
+ * termination. atexit functions will be executed prior to oe_allocator_cleanup.
+ * oe_allocator_cleanup may not register functions with atexit.
```

Comment:
Similarly, consider simplifying:
```suggestion
 * This function will be called by oecore during enclave termination.
 * atexit functions will be executed prior to oe_allocator_cleanup.
 * oe_allocator_cleanup may not register functions with atexit.
```

Reply:
Done.